packages/eslint-plugin/docs/rules/return-await.mdx

```diff
@@ -9,12 +9,14 @@ import TabItem from '@theme/TabItem';
 >
 > See **https://typescript-eslint.io/rules/return-await** for documentation.
 
-Returning an awaited promise can make sense for better stack trace information as well as for consistent error handling (returned promises will not be caught in an async function try/catch).
-
 This rule builds on top of the [`eslint/no-return-await`](https://eslint.org/docs/rules/no-return-await) rule.
```

Comment:
i kinda think we should drop this, as well as the comments around the naming. the base rule has been deprecated for some time, and this rule has an independent life of its own now. 

I think, instead, it would be better to relegate the reference to the base rule to a little info box saying (approximaely) "this was originally conceived as an extension rule associated with the eslint rule `no-return-await`, but that's no longer the case. The base rule became deprecated because it was realized that it did more harm than good, due to limitations of not having access to type information. If you're interested in reading on the history of that decision, see \<insert link\>"

What do you think?

Reply:
FWIW, the base rule was deprecated because it was based on the false premise that `return promise;` is faster than `return await promise;`, which may be true for some time but not anymore. `return await promise;` is desirable in all other regards. It's totally possible to build a type-unaware rule that changes `return await something;` to `return something;`, but not in the other direction.

I think we should keep it at least for another ESLint major version, as people are still familiar with this rule.

Reply:
> FWIW, the base rule was deprecated because it was based on the false premise that `return promise;` is faster than `return await promise;`

Right, performance was the once-believed-but-now-debunked good.

> `return await promise;` is desirable in all other regards. It's totally possible to build a type-unaware rule that changes `return await something;` to `return something;`, but not in the other direction.

harm, due to limitations of not having access to type information.

***

That's where I get my summary that it was doing more harm than good due to the limitations of not having access to type information 🙂 But no need to use this wording if you feel otherwise!